RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+However, with an approach which provides for this complexity, we can continue to simplify the experience for our users to from:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell
+```
+
+to
+
+```powershell
+PS> docker run mcr.microsoft.com/powershell
+```
+
+or
+
+```powershell
+PS> Docker image ls --format='{{json .}}' | ConvertFrom-Json | Where-Object {$_.Repository -match "mcr"}| Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+to this:
+
+```powershell
+PS> docker image ls | ? Repository -match "mcr"
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+### Follow-on Work
+
+A number of follow-on activities should be considered to bring continued improvement of the native command experience.
+
+#### Improved TabCompletion
+
+Current tab-completion (on Unix) is done via a somewhat fragile call to the native shell.
+Tab-completion on Windows must be implemented specifically.
+we should take the opportunity to improve our tooling around tab-completion to make it easier to create,
+and provide generalized help parsers (where possible) to generate tab-completion.
+`bash` provides a number of more generalized tab-completion, we should attempt to do the same for PowerShell users.
+
+For example, bash has a single completer all known hosts which is used by nearly 20 executables.
+We already will tab-complete for file system items,
+a small effort should be able to increase our coverage for other native executables.
+
+#### Improved Formatting Tooling
+
+Once text output is converted to objects,
+we should improve the process for creating formatting directives.
+When creating a formatting file,
+the user needs to manually author the `XML` and then call `update-formatdata` to add that configuration to the current session.
+Usually, formatting files are associated with a module and distributed in this way.
+With stand-alone native applications, it means that providing for reasonable formatting will require an additional file.
+A more dynamic approach for formatting would improve the experience for the user.
+
+## Open Questions
+
+### History
+
+It's not clear whether history should be altered to show the added pipeline elements. We don't currently do that for `out-default` which can also add to the pipeline.
+It should be relatively easy to determine whether the pipeline as the `app-json`, as `get-command app` will include the name of the json adapter.
```

Comment:
I'm of the opinion we just save what is typed by the user and not added pipeline elements to avoid confusion. However this could be another case for adding additional fields to history objects so show what is added to the pipeline.